HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -1991,16 +1991,41 @@ def update(self):
 
         Parameters
         ----------
-        none
+        None
 
         Returns
         -------
-        none
+        None
         '''
         self.updateIncomeProcess()
         self.updateAssetsGrid()
         self.updateSolutionTerminal()
+        
+        
+    def resetRNG(self):
+        '''
+        Reset the RNG behavior of this type.  This method is called automatically
+        by initializeSim(), ensuring that each simulation run uses the same sequence
+        of random shocks; this is necessary for structural estimation to work.
+        This method extends AgentType.resetRNG() to also reset elements of IncomeDstn.
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        None
+        '''
+        PerfForesightConsumerType.resetRNG(self)
+        
+        # Reset IncomeDstn if it exists (it might not because resetRNG is called at init)
+        if hasattr(self, 'IncomeDstn'):
+            T = len(self.IncomeDstn)
+            for t in range(T):
+                self.IncomeDstn[t].reset()
```

Comment:
possible rewrite:

for distribution in self.IncomeDstn:
  distribution.reset()